lib/types/bcachefs_filesystem.nix

```diff
@@ -301,38 +327,130 @@
       internal = true;
       readOnly = true;
       default =
-        (lib.optional (config.mountpoint != null) {
-          fileSystems.${config.mountpoint} = {
-            device = "/dev/disk/by-uuid/${config.uuid}";
-            fsType = "bcachefs";
-            options = lib.unique ([ "X-mount.mkdir" ] ++ config.mountOptions);
-            neededForBoot = true;
-          };
-        })
-        ++ (map (subvolume: {
-          fileSystems.${subvolume.mountpoint} = {
-            device = "/dev/disk/by-uuid/${config.uuid}";
-            fsType = "bcachefs";
-            options = lib.unique (
+        if pkgs == null then
+          [ ]
+        else
+          (lib.optional (config.mountpoint != null) {
+            fileSystems.${config.mountpoint} = {
+              device = "/dev/disk/by-uuid/${config.uuid}";
+              fsType = "bcachefs";
+              options = lib.unique ([ "X-mount.mkdir" ] ++ config.mountOptions);
+              neededForBoot = true;
+            };
+          })
+          ++ (map (subvolume: {
+            fileSystems.${subvolume.mountpoint} = {
+              device = "/dev/disk/by-uuid/${config.uuid}";
+              fsType = "bcachefs";
+              options = lib.unique (
+                [
+                  "X-mount.mkdir"
+                  "X-mount.subdir=${lib.removePrefix "/" subvolume.name}"
+                ]
+                ++ subvolume.mountOptions
+              );
+              neededForBoot = true;
+            };
+          }) (lib.filter (subvolume: subvolume.mountpoint != null) (lib.attrValues config.subvolumes)))
+          ++ (lib.optional (config.unlock.enable) {
+            # Dependencies for Clevis unlocking
+            boot.initrd.extraPackages =
+              with pkgs;
               [
-                "X-mount.mkdir"
-                "X-mount.subdir=${lib.removePrefix "/" subvolume.name}"
+                clevis
```

Comment:
AFAIK clevis does not have FIDO2 support: https://github.com/latchset/clevis/pull/399